src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java

```diff
@@ -0,0 +1,84 @@
+ * limitations under the License.
+ */
+package org.apache.commons.io;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * This class ensure the correctness of {@link FileUtils#copyDirectoryToDirectory(File, File)} (File,File)}.
+ * TODO: currently does not cover happy cases
+ *
+ * @see FileUtils#copyDirectoryToDirectory(File, File)
+ */
+public class FileUtilsCopyDirectoryToDirectoryTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void copyDirectoryToDirectoryThrowsIllegalExceptionWithCorrectMessageWhenSrcDirIsNotDirectory() throws IOException {
+        File srcDir = temporaryFolder.newFile("notadirectory");
+        File destDir = temporaryFolder.newFolder("destinationDirectory");
+        String expectedMessage = String.format("Source '%s' is not a directory", srcDir);
+        assertExceptionTypeAndMessage(srcDir, destDir, IllegalArgumentException.class, expectedMessage);
+    }
+
+    @Test
+    public void copyDirectoryToDirectoryThrowsIllegalArgumentExceptionWithCorrectMessageWhenDstDirIsNotDirectory() throws IOException {
+        File srcDir = temporaryFolder.newFolder("sourceDirectory");
+        File destDir =  temporaryFolder.newFile("notadirectory");
+        String expectedMessage = String.format("Destination '%s' is not a directory", destDir);
+        assertExceptionTypeAndMessage(srcDir, destDir, IllegalArgumentException.class, expectedMessage);
+    }
+
+    @Test
+    public void copyDirectoryToDirectoryThrowsNullPointerExceptionWithCorrectMessageWhenSrcDirIsNull() throws IOException {
+        File srcDir = null;
+        File destinationDirectory =  temporaryFolder.newFolder("destinationDirectory");
+        String expectedMessage = "Source must not be null";
+        assertExceptionTypeAndMessage(srcDir, destinationDirectory, NullPointerException.class,  expectedMessage);
+    }
+
+    @Test
+    public void copyDirectoryToDirectoryThrowsNullPointerExceptionWithCorrectMessageWhenDstDirIsNull() throws IOException {
+        File srcDir = temporaryFolder.newFolder("sourceDirectory");
+        File destDir =  null;
+        String expectedMessage = "Destination must not be null";
+        assertExceptionTypeAndMessage(srcDir, destDir, NullPointerException.class, expectedMessage);
+    }
+
+    private static void assertExceptionTypeAndMessage(File srcDir, File destDir, Class expectedExceptionType, String expectedMessage) {
+        try {
+            FileUtils.copyDirectoryToDirectory(srcDir, destDir);
+        } catch (Exception e) {
+            String msg = e.getMessage();
+            assertEquals(expectedExceptionType, e.getClass());
+            assertEquals(expectedMessage, msg);
+            return;
+        }
+        fail();
+
```

Comment:
Remove extra line.

Reply:
Fixed in https://github.com/apache/commons-io/pull/95/commits/a3a0bfa102f7b304c067d670a7bf32aa45863a6e